url.bs

```diff
@@ -1861,6 +1859,8 @@ these steps:
    <dt><dfn>relative state</dfn>
    <dd>
     <ol>
+     <li><p>Assert: <var>base</var>'s <a for=url>scheme</a> is not "<code>file</code>".
```

Comment:
One can only reach relative state from no scheme state and special relative or authority state.

[No scheme state](https://url.spec.whatwg.org/#no-scheme-state) can only switch to relative state if base’s scheme is not "file" (step 3).

[Special relative or authority state](https://url.spec.whatwg.org/#special-relative-or-authority-state) is only reachable from [scheme state](https://url.spec.whatwg.org/#scheme-state), which rules out "file" scheme in step 2.5 but switches to relative state in step 2.6.